Swift/tree.playground/Contents.swift

```diff
@@ -0,0 +1,100 @@
+import Foundation
+
+
+class Node<T>: CustomStringConvertible where T: CustomStringConvertible, T: Comparable {
+    
+    var value: T?
+    weak var parentNode: Node?
+    var subNodes = [Node]()
+    var subNodesCount: Int {
+        return subNodes.count
+    }
+    var isEmpty: Bool {
+        return subNodes.isEmpty
+    }
+    
+    init() { }
+    
+    init(value: T) {
+        self.value = value
+    }
+    
+    func addChild(node child: Node) {
+        child.parentNode = self
+        subNodes.append(child)
+    }
+    
+    func delete() {
+        parentNode!.subNodes = parentNode!.subNodes.filter({ $0 !== self })
```

Comment:
That's a very bad idea combined with using the value of ``parentNode`` as a ``Node?`` with the write access from outside the class.

If the user tries for some reason doing the following
```Swift
let tree = Tree<Int>()
tree.root.value = 99
tree.root.delete()
```
The result is predictable.